ide/coqide/wg_ScriptView.ml

```diff
@@ -575,19 +598,21 @@ object (self)
 
     (* todo: gray out cut, paste, undo, replace, [un]comment, Templates in the
       menu when script is not editable *)
-    let nonmod_keys = List.map (fun k -> let (key, _) = GtkData.AccelGroup.parse k in key)
-      [ "Left"; "Right"; "Up"; "Down"; "Home"; "End"; "<Ctrl>A"] in
+    let nonmod_keys = List.map (fun k -> GtkData.AccelGroup.parse k)
+      [ "Left"; "Right"; "Up"; "Down"; "Home"; "End"; "<Ctrl>A";
```

Comment:
I am afraid that this hack is really nonsense. For instance, it prevents moving through the file when busy using page up / page down characters. We shouldn't look for keycodes to check whether we have to prevent the keypress to have any effect.

Reply:
@jfehrle did you try setting the processed tag to be uneditable via a signal? If this works this is the normal way to do this in GTK.

Reply:
Page up/down was an oversight.  I wasn't aware of the ``` `EDITABLE``` tag until after I wrote the code.

> did you try setting the processed tag to be uneditable via a signal? If this works this is the normal way to do this in GTK.did you try setting the processed tag to be uneditable via a signal? If this works this is the normal way to do this in GTK.

You mean tagging the processed text with ``` `EDITABLE```?  What signal mechanism are you referring to (e.g. name of the API)?  Not so easy to figure out what GTK offers and how to use it from OCaml.

Reply:
There is a documentation of GTK and the API of the lablgtk bindings which are systematically derived from it...